src/phoenix/server/api/mutations/api_key_mutations.py

```diff
@@ -43,51 +42,52 @@ class DeleteSystemApiKeyMutationPayload:
 
 @strawberry.type
 class ApiKeyMutationMixin:
-    @strawberry.mutation(permission_classes=[HasSecret, IsAuthenticated])  # type: ignore
+    @strawberry.mutation(
+        permission_classes=[
+            IsNotReadOnly,
+            HasSecret,
+            IsNotReadOnly,
+            IsAuthenticated,
+            IsAdmin,
+        ]
+    )  # type: ignore
     async def create_system_api_key(
         self, info: Info[Context, None], input: CreateApiKeyInput
     ) -> CreateSystemApiKeyMutationPayload:
-        # TODO(auth): safe guard against auth being disabled and secret not being set
+        assert (token_store := info.context.token_store) is not None
+        user_role = "SYSTEM"
         async with info.context.db() as session:
             # Get the system user - note this could be pushed into a dataloader
             system_user = await session.scalar(
                 select(models.User)
                 .join(models.UserRole)  # Join User with UserRole
-                .where(models.UserRole.name == "SYSTEM")  # Filter where role is SYSTEM
+                .where(models.UserRole.name == user_role)  # Filter where role is SYSTEM
+                .order_by(models.User.id)
                 .limit(1)
             )
             if system_user is None:
                 raise ValueError("System user not found")
-
-            insert_stmt = (
-                insert(models.APIKey)
-                .values(
-                    user_id=system_user.id,
-                    name=input.name,
-                    description=input.description or None,
-                    expires_at=input.expires_at or None,
-                )
-                .returning(models.APIKey)
-            )
-            api_key = await session.scalar(insert_stmt)
-            assert api_key is not None
-
-        encoded_jwt = create_jwt(
-            secret=info.context.get_secret(),
-            name=api_key.name,
-            id=api_key.id,
-            description=api_key.description,
-            iat=api_key.created_at,
-            exp=api_key.expires_at,
+        issued_at = datetime.now(timezone.utc)
+        claims = Claim(
+            user_id=system_user.id,
+            issuer=Issuer.API_KEY,
```

Comment:
this doesn't seem right according to how I understand issuer:

> [4.1.1](https://www.rfc-editor.org/rfc/rfc7519#section-4.1.1).  "iss" (Issuer) Claim

   The "iss" (issuer) claim identifies the principal that issued the
   JWT.  The processing of this claim is generally application specific.
   The "iss" value is a case-sensitive string containing a StringOrURI
   value.  Use of this claim is OPTIONAL.

Reply:
Our usage seems compatible with this sentence (in my interpretation). What is your understanding of it?

> The "iss" (issuer) claim identifies the principal that issued the JWT. 
The processing of this claim is generally application specific.

Reply:
but the issuer in this case always the phoenix server no? Maybe I'm less familiar with this application specific `iss` definition. Do you have a link to some literature I can view on this?

Reply:
The section in the OAuth document actually also doesn't say anything concrete about what this claim should be, but what it does make clear is that how it can be used:

> If the value does not match the expected issuer identifier, clients MUST reject the authorization response and MUST NOT proceed with the authorization grant.

Given this reading i think we can draw the following conclusions.
- `iss` can be any value, as long as it fulfills its intent. 
- the concept of `iss` is not relevant for us, because nothing in our code requires validating an `iss` (which is true in general for any kind of _internal_ process)

With that said, our usage is also not contradictory because it's a plain-text interpretation of the JWT RFC, namely leaning on the keyword "identifies", which also is consistent with the application in the OAuth document (minus the "matching" step):

> The "iss" (issuer) claim identifies the principal that issued the JWT.

Additionally, in order for "matching" step to work as described the OAuth doc, expected values must be present, which are by definition _internal_ values. Our usage here is just exactly that: internal values.

All that is to say: we can rename it if you want, haha (because either way it doesn't matter).